src/Servers/Kestrel/test/InMemory.FunctionalTests/ChunkedResponseTests.cs

```diff
@@ -529,6 +530,8 @@ public async Task WritesAreFlushedPriorToResponseCompletion()
         [Fact]
         public async Task ChunksCanBeWrittenManually()
         {
+            AppContext.SetSwitch("Switch.Microsoft.AspNetCore.Server.Kestrel.DisableAutoChunk", true);
```

Comment:
Calling SetSwitch in tests doesn't work well, it's global and causes side effects for parallel tests. It's better to have the product code check the switch in the constructor, store it in an internal variable, and I have the test directly set the internal variable.

Reply:
How can I set the internal variable on `HttpProtocol` from this test?